backend/middleware/multer.js

```diff
@@ -0,0 +1,29 @@
+import multer from "multer";
+
+// uploaded files will be stored on the local disk and saved with the file name
+const storage = multer.diskStorage({
+  filename: (req, file, cb) => {
+    cb(null, file.originalname);
+  },
+});
+
+const upload = multer({
+  storage: storage,
+  fileFilter: (req, file, cb) => {
+    const allowedMimeTypes = [
+      "image/jpeg", 
+      "image/png", 
+      "image/gif",
+      "video/mp4",
+      "video/webm",
+      "video/avi"
+    ];
+    if (allowedMimeTypes.includes(file.mimetype)) {
+      cb(null, true);
+    } else {
+      cb(new Error("Invalid file type. Only JPEG, PNG, GIF, MP4, WebM, and AVI are allowed."));
```

Comment:
This error msg didnt really go through to frontend i believe.
When trying to add an image to a locket regarding this, i accidentally tried to upload a pdf but just got an alert with a fairly generic "Failed to upload"